Codigos/Estruturas-de-Dados/XorTrie/XorTrie.cpp

```diff
@@ -0,0 +1,63 @@
+template <int bits = 30> struct XorTrie {
+    vector<vector<int>> go;
+    int root, cnt;
+
+    int new_node() {
+        go[0][cnt] = -1;
+        go[1][cnt] = -1;
+        return cnt++;
+    }
+
+    void insert(int x) {
+        int v = root;
+        for (int i = bits - 1; i >= 0; i--) {
+            if (go[x >> i & 1][v] == -1) {
+                go[x >> i & 1][v] = new_node();
+            }
+            v = go[x >> i & 1][v];
```

Comment:
WDYT of putting `x >> i & 1` in a variable?
```suggestion
            int bit = x >> i & 1;
            if (go[bit][v] == -1) {
                go[bit][v] = new_node();
            }
            v = go[bit][v];
```